src/NLog/LayoutRenderers/Wrappers/ReplaceNewLinesLayoutRendererWrapper.cs

```diff
@@ -64,33 +62,51 @@ public string Replacement
             set
             {
                 _replacement = value;
-                _replaceWithNewLines = value?.IndexOf('\n') >= 0;
             }
         }
         private string _replacement = " ";
-        private bool _replaceWithNewLines;
 
         /// <summary>
         /// Gets or sets a value indicating the string that should be used to replace newlines.
         /// </summary>
         /// <docgen category='Layout Options' order='50' />
         public string ReplaceNewLines { get => Replacement; set => Replacement = value; }
 
+        private static readonly char[] LineEndCharacters = new char[] { '\u000D', '\u000A', '\u0085', '\u2028', '\u000C', '\u2029' };
```

Comment:
Instead of using unicode-values directly, then I think it would be easier if you used standard chars when possible. Ex `\n` + `\r` + etc.